drivers/platform/altera/i2c.c

```diff
@@ -89,15 +89,13 @@ int32_t i2c_remove(struct i2c_desc *desc)
  * @param desc - The I2C descriptor.
  * @param data - Buffer that stores the transmission data.
  * @param bytes_number - Number of bytes to write.
- * @param option - Stop condition control.
- *                   Example: 0 - A stop condition will not be generated;
- *                            1 - A stop condition will be generated.
+ * @param transfer_mode - Transfer mode type
  * @return SUCCESS in case of success, FAILURE otherwise.
  */
 int32_t i2c_write(struct i2c_desc *desc,
 		  uint8_t *data,
 		  uint8_t bytes_number,
-		  uint8_t option)
+		  enum i2c_transfer_mode transfer_mode)
```

Comment:
I think that because i2c_transfer_mode is defined as a typedef, you can skip the `enum` qualifier completely so your parameter could be just:
```
...,
i2c_transfer_mode transfer_mode)
```

Reply:
According to linux coding style we mustn't use typedefs. I wanted to remove the typedef but I'm not sure if some other file use the structure names without the struct keyword.